lib/compiler/passes/generate-js.js

```diff
@@ -383,7 +383,7 @@ function generateJS(ast, options) {
       "          break;",
       "",
       "        default:",
-      "          throw new Error(\"Invalid opcode: \" + bc[ip] + \".\");",
+      "          throw new Error(\"Invalid opcode #\" + ip + \": \" + bc[ip] + \".\");",
```

Comment:
I’m curious: you apparently missed the opcode position in the message but you didn’t miss an indication in bytecode of *which rule* the invalid opcode was? I’d rather add both, e.g. “Invalid opcode: 77 (rule: 5, position: 6).”.